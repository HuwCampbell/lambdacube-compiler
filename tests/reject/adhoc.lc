
data A = A

x = oneComp :: Vec 4 A

main = x    -- TODO: catch error at application


