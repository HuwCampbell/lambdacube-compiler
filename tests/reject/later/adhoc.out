during inst x
position: "./tests/reject/adhoc.lc" (line 4, column 1) - "./tests/reject/adhoc.lc" (line 4, column 23)
x = oneComp :: Vec 4 A
^^^^^^^^^^^^^^^^^^^^^^
during ambcheck x ambiguous type:
(a :: Component (Vec 4 A)) => Vec 4 A
problematic vars: [(t00004,Component (Vec 4 A))]